### DISCUSSION TITLE
ScriptReference/Component.GetComponent

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#141] [Notes]
# ScriptReference/Component.GetComponent

https://docs.unity3d.com/ScriptReference/Component.GetComponent.html

**Comment by vertxxyz** [comment]
To handle the case where a component may not exist, use [`TryGetComponent`](https://docs.unity3d.com/ScriptReference/Component.TryGetComponent.html) instead.
In the editor `GetComponent` will allocate if the component is not found because the returned [Unity null](https://help.vertx.xyz/programming/other/unity-null) contains extra information to help debugging in the editor.
